modules/api-gateway/main.tf

```diff
@@ -0,0 +1,227 @@
+}
+
+resource "aws_api_gateway_method" "orca_catalog_reporting_api_method" {
+  rest_api_id = aws_api_gateway_rest_api.orca_cumulus_reconciliation_api.id
+  resource_id = aws_api_gateway_resource.orca_catalog_reporting_api_resource.id
+  http_method = "ANY"
+  # todo: Make sure this is locked down against external access.
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "orca_catalog_reporting_api_integration" {
+  rest_api_id             = aws_api_gateway_rest_api.orca_cumulus_reconciliation_api.id
+  resource_id             = aws_api_gateway_resource.orca_catalog_reporting_api_resource.id
+  http_method             = aws_api_gateway_method.orca_catalog_reporting_api_method.http_method
+  integration_http_method = "POST"
+  type                    = "AWS"
+  uri                     = var.orca_catalog_reporting_invoke_arn
+}
+
+resource "aws_api_gateway_method_response" "orca_catalog_reporting_response_200" {
+  rest_api_id = aws_api_gateway_rest_api.orca_cumulus_reconciliation_api.id
+  resource_id = aws_api_gateway_resource.orca_catalog_reporting_api_resource.id
+  http_method = aws_api_gateway_method.orca_catalog_reporting_api_method.http_method
+  status_code = "200"
+}
+
+resource "aws_api_gateway_integration_response" "orca_catalog_reporting_api_response" {
+  depends_on  = [aws_api_gateway_integration.orca_catalog_reporting_api_integration]
+  rest_api_id = aws_api_gateway_rest_api.orca_cumulus_reconciliation_api.id
+  resource_id = aws_api_gateway_resource.orca_catalog_reporting_api_resource.id
+  http_method = aws_api_gateway_method.orca_catalog_reporting_api_method.http_method
+  status_code = aws_api_gateway_method_response.orca_catalog_reporting_response_200.status_code
+  # Transforms the backend JSON response to XML. Currently being used by create_http_error_dict() function in orca_catalog_reporting.py
+  response_templates = {
+    "application/json" = <<EOF
+    #set($inputRoot = $input.path('$'))
+    $input.json("$")
+    #if($input.path("stackTrace") != '')
+    #set($context.responseOverride.status = 500)
+    #elseif($input.path("httpStatus") != '')
+    #set($context.responseOverride.status = $input.path('httpStatus'))
+    #end
+    EOF
+  }
+}
+
+resource "aws_lambda_permission" "orca_catalog_reporting_api_permission" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = "${var.prefix}_orca_catalog_reporting"
+  principal     = "apigateway.amazonaws.com"
+  source_arn    = "${aws_api_gateway_rest_api.orca_cumulus_reconciliation_api.execution_arn}/*/${aws_api_gateway_method.orca_catalog_reporting_api_method.http_method}${aws_api_gateway_resource.orca_catalog_reporting_api_resource.path}"
+}
+
+# API details for  for request_status_for_granule lambda function
+resource "aws_api_gateway_resource" "request_status_for_granule_api_resource" {
+  path_part   = "recovery_status_granules"
```

Comment:
Should probably discuss this further. I think we would want a `path` instead of `path_part` and we would want to follow RESTFUL API standards.